### PR TITLE
fix social connection issues

### PR DESCRIPTION
**Pull request by sgfost**
- Profile edit form redirects to /accounts/social/connections as a central form (which now correctly uses POST) for managing account connections which looks like:
Screenshot from 2023-02-27 12-53-12
alternatively, could render the direct links in the template before mounting the vue component form

- errors are now display properly in the connections page instead of rendering a method
- links to password reset page when trying to remove a lone connection from an account with no password set

### CODE REVIEW

File: frontend/src/components/profile/Edit.vue [hunk 1]
```diff
@@ -35,22 +35,30 @@
...
+              </a>
+              <a title="Edit connection" :href="socialConnectionsURL">
+                <i class="float-right fas fa-edit"></i>
+              </a>
             </span>
             <span v-else>
-              <a title="orcid" href="/accounts/orcid/login/?process=connect"
-                ><span class="text-gray fab fa-orcid"></span> Connect your ORCID account</a
-              >
+              <a title="github" :href="socialConnectionsURL">
```
Comment (alee) [inline frontend/src/components/profile/Edit.vue:46]: We could either get rid of the  `a title` here and elsewhere in this text or add more semantic info to them like "sign in with orcid", "sign in with github" etc

File: django/core/jinja2/socialaccount/connections.jinja [hunk 2]
```diff
@@ -37,6 +37,13 @@
             </div>
         </div>
         <button type="submit" class='btn btn-sm btn-primary'><span class='fa fa-trash'></span> Remove</button>
+        {% if form.accounts|length == 1 and not request.user.has_usable_password() %}
```
Comment (alee) [inline django/core/jinja2/socialaccount/connections.jinja:40]: good catch!

File: django/core/jinja2/socialaccount/connections.jinja [hunk 3]
```diff
@@ -19,7 +19,7 @@
         {{ csrf_input }}
 
         {% if form.non_field_errors() %}
-            <div id="errorMsg">{{ form.non_field_errors }}</div>
+            <div id="errorMsg">{{ form.non_field_errors() }}</div>
```
Comment (alee) [inline django/core/jinja2/socialaccount/connections.jinja:22]: :100:

**Review by alee — COMMENTED** [review]
LGTM, could fix the <a href title=...> attributes or leave for another day

File: frontend/src/components/profile/Edit.vue (hunk 1 above)
Comment (sgfost) [reply frontend/src/components/profile/Edit.vue]: cleaned up the markup around here + some other consistency changes

**Review by alee — APPROVED** [review]
Tested on staging - LGTM, thanks for this @sgfost !

**Pull request merged**